lib/protein/adapters/amqp_adapter/server.ex

```diff
@@ -102,26 +125,27 @@ defmodule Protein.AMQPAdapter.Server do
 
   defp consume(chan, opts, meta, payload) do
     server_mod = Keyword.fetch!(opts, :server_mod)
-
-    {response, error} = try_process(payload, server_mod)
-
+    response = server_mod.process(payload)
     if should_respond(response, meta), do: respond(response, chan, meta)
+    Basic.ack(chan, meta.delivery_tag)
+  end
 
+  def handle_consumer_error(chan, meta) do
+    if should_respond("ESRV", meta), do: respond("ESRV", chan, meta)
     Basic.ack(chan, meta.delivery_tag)
+  end
 
-    if error do
-      {exception, stacktrace} = error
-      reraise(exception, stacktrace)
-    end
+  def kill_consumers(consumers) do
+    Enum.each(consumers, fn %{pid: pid, monitor_ref: ref} ->
+      Process.demonitor(ref)
+      Process.exit(pid, :kill)
+    end)
   end
 
-  defp try_process(payload, server_mod) do
-    response = server_mod.process(payload)
-    {response, nil}
-  rescue
-    exception ->
-      stacktrace = System.stacktrace()
-      {"ESRV", {exception, stacktrace}}
+  def get_consumer(consumers, searched_pid) do
+    [consumer] = Enum.filter(consumers, fn %{pid: pid} -> pid == searched_pid end)
```

Comment:
Looking at how we're using these consumers it seems that `Map` using pid as key would be a better structure for it (e.g. it has `Map.pop`).